packages/nodes-base/nodes/Google/CloudStorage/BucketDescription.ts

```diff
@@ -0,0 +1,511 @@
+							let executions: INodeExecutionData[] = [];
+							let responseData: INodeExecutionData[];
+							let nextPageToken: string | undefined = undefined;
+
+							do {
+								requestOptions.options.qs.pageToken = nextPageToken;
+								responseData = await this.makeRoutingRequest(requestOptions);
+
+								// Check for another page
+								const lastItem = responseData[responseData.length - 1].json;
+								nextPageToken = lastItem.nextPageToken as string | undefined;
+
+								// Extract just the list of buckets from the page data
+								responseData.forEach((page) => {
+									const buckets = page.json.items as IDataObject[];
+									executions = executions.concat(buckets.map((bucket) => ({ json: bucket })));
+								});
+							} while (nextPageToken);
+
+							// Return all execution responses as an array
+							return executions;
+						},
+					},
+				},
+				action: 'Get all Buckets',
+			},
+			{
+				name: 'Update',
+				value: 'update',
+				description: 'Update the metadata of a bucket',
+				routing: {
+					request: {
+						method: 'PUT',
+						url: '={{"/b/" + $parameter["bucketName"]}}',
+						qs: {
+							project: '={{$parameter["project"]}}',
+						},
+						body: {},
+						returnFullResponse: true,
+					},
+					send: {
+						preSend: [
+							async function (this, requestOptions) {
+								if (!requestOptions.qs) requestOptions.qs = {};
+								if (!requestOptions.body) requestOptions.body = {};
+								const additionalQs = this.getNodeParameter('createAcl') as IDataObject;
+								const additionalBody = this.getNodeParameter('createBody') as IDataObject;
+								const filters = this.getNodeParameter('getFilters') as IDataObject;
+
+								// Merge in the options into the queryset
+								requestOptions.qs = Object.assign(requestOptions.qs, additionalQs, filters);
+								requestOptions.body = Object.assign(requestOptions.body, additionalBody);
+								return requestOptions;
+							},
+						],
+					},
+				},
+				action: 'Create a new Bucket',
+			},
+		],
+		default: 'getAll',
+	},
+];
+
+export const bucketFields: INodeProperties[] = [
+	{
+		displayName: 'Project',
+		name: 'project',
+		type: 'string',
+		required: true,
+		placeholder: 'Project Name',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'getAll'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Bucket Name',
+		name: 'bucketName',
+		type: 'string',
+		placeholder: 'Bucket Name',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['create', 'get', 'update', 'delete'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Prefix',
+		name: 'prefix',
+		type: 'string',
+		placeholder: 'Filter for Bucket Names',
+		displayOptions: {
+			show: {
+				resource: ['bucket'],
+				operation: ['getAll'],
+			},
+		},
+		default: '',
+	},
+	{
+		displayName: 'Projection',
```

Comment:
I don't think this property is doing anything. Could you confirm?

Reply:
This parameter is an optional one that basically says whether the response should contain access control data or not basically!

https://cloud.google.com/storage/docs/json_api/v1/buckets/list can see it here for example